src/core.ts

```diff
@@ -69,9 +69,8 @@ export function getSanitizedEnv<S>(
     // If no value was given and default/devDefault were provided, return the appropriate default
     // value without passing it through validation
     if (rawValue === undefined) {
-      // Use devDefault values only if NODE_ENV was explicitly set, and isn't 'production'
       const usingDevDefault =
-        rawNodeEnv && rawNodeEnv !== 'production' && spec.hasOwnProperty('devDefault')
+         rawNodeEnv !== 'production' && spec.hasOwnProperty('devDefault')
```

Comment:
I think this change causes a regression unfortunately– when NODE_ENV isn't provided, envalid assumes "production" mode. It's the safest choice in case for some reason your prod env doesn't have NODE_ENV set